### PR TITLE
setting auth as two different pwds

### PR DESCRIPTION
**Pull request by tcplglo**
Basic evaluation of the connector's authentication password allowing rotation.

Initially, this could be just set with two env vars rather than complicating things with a JSON string?

### CODE REVIEW

**Review by DownChapel — COMMENTED** [review]
I think we need to update the API extension with the `basicAuthPwdCurrent` if the API 
extension already exists (in the post-deploy action)?

Otherwise if the `BASIC_AUTH_PASSWORD_CURRENT` value changes, the API extension will still be using the old value. This would continue working if the `BASIC_AUTH_PASSWORD_PREVIOUS` value was set, but then if you removed the previous setting it would break.

Be good to test this in connect

**Comment by tcplglo** [comment]
> I think we need to update the API extension with the `basicAuthPwdCurrent` if the API extension already exists (in the post-deploy action)?
> 
> Otherwise if the `BASIC_AUTH_PASSWORD_CURRENT` value changes, the API extension will still be using the old value. This would continue working if the `BASIC_AUTH_PASSWORD_PREVIOUS` value was set, but then if you removed the previous setting it would break.
> 
> Be good to test this in connect

{A5143719-2806-4A37-BC9D-149DB820395C}

we do not get a PUT option just a POST...

**Comment by DownChapel** [comment]
> we do not get a PUT option just a POST...

Yeah, they use POST for updates too https://docs.commercetools.com/api/projects/api-extensions#update-extension

File: README.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ To install the connector you will need to provide the following configuration se
 - `DOVETECH_API_HOST` - this is available on the Details tab of the Dovetech Customer Portal
 - `DOVETECH_API_KEY` - this is the Processor API Key from your project in the Dovetech Customer Portal
 - The commercetools settings prefixed with `CTP_` should be populated from the API client created above
+- `BASIC_AUTH_PASSWORD` - these two variables (current/previous) will allow the connector to validate that calls come from commercetools. `BASIC_AUTH_PASSWORD_CURRENT` is required, and has a minimum length of two characters and a maximum of 50. The contents will will be hashed for evaluation.
```
Comment (DownChapel) [inline README.md]: Maybe remove the comment about hashing? It's not strictly hashed as I understand it - it's base64 encoded and then creates a buffer?

```suggestion
- `BASIC_AUTH_PASSWORD` - these two variables (current/previous) will allow the connector to validate that calls come from commercetools. `BASIC_AUTH_PASSWORD_CURRENT` is required, and has a minimum length of two characters and a maximum of 50.
```

File: cart-service/src/validators/env.validators.ts [hunk 2]
```diff
@@ -48,9 +49,9 @@ const envValidators = [
     referencedBy: 'environmentVariables',
   }),
 
-  standardString(['connectorBasicAuthPassword'],{
-    code: 'connectorBasicAuthPassword',
-    message: 'Basic auth password should be a valid string',
+  longString(['basicAuthPwdCurrent'],{
```
Comment (DownChapel) [inline cart-service/src/validators/env.validators.ts]: Might be better to use the `standardString` function and pass in the min and max values using `overrideConfig` like `clientSecret` does?

**Review by DownChapel — COMMENTED** [review]
Looks good. Couple of minor comments

File: README.md (hunk 1 above)
Comment (tcplglo) [reply README.md]: removed

File: cart-service/src/validators/env.validators.ts (hunk 2 above)
Comment (tcplglo) [reply cart-service/src/validators/env.validators.ts]: updated

**Review by DownChapel — APPROVED** [review]

**Pull request merged**